packages/dx-grid-core/src/plugins/virtual-table-state/helpers.test.ts

```diff
@@ -228,10 +229,45 @@ describe('VirtualTableState helpers', () => {
       const virtualRows = createVirtualRows(loadedInterval);
 
       it('should return prev, current and next pages', () => {
-        expect(calculateRequestedRange(virtualRows, newInterval, 1170, pageSize))
+        expect(calculateRequestedRange(virtualRows, newInterval, pageSize))
           .toEqual({ start: 1000, end: 1300 });
       });
     });
+
+    describe('reference index', () => {
+      // tslint:disable-next-line: max-line-length
+      it('should caclulate correct if reference index less than the start of a new interval', () => {
+        const loadedInterval = createInterval(100, 400);
+        const newInterval = createInterval(200, 500);
+        const virtualRows = createVirtualRows(loadedInterval);
+        const referenceIndex = 440;
+
+        expect(calculateRequestedRange(virtualRows, newInterval, pageSize, referenceIndex))
+          .toEqual({ start: 400, end: 500 });
+      });
+
+      // tslint:disable-next-line: max-line-length
+      it('should caclulate correct if reference index more than the start of a new interval and less than half of page', () => {
+        const loadedInterval = createInterval(100, 400);
+        const newInterval = createInterval(200, 500);
+        const virtualRows = createVirtualRows(loadedInterval);
+        const referenceIndex = 360;
+
+        expect(calculateRequestedRange(virtualRows, newInterval, pageSize, referenceIndex))
+          .toEqual({ start: 400, end: 500 });
+      });
+
+      // tslint:disable-next-line: max-line-length
+      it('should caclulate correct if reference index more than the start of a new interval and less than half of page', () => {
+        const loadedInterval = createInterval(100, 400);
+        const newInterval = createInterval(200, 500);
+        const virtualRows = createVirtualRows(loadedInterval);
+        const referenceIndex = 320;
+
+        expect(calculateRequestedRange(virtualRows, newInterval, pageSize, referenceIndex))
+          .toEqual({ start: 300, end: 400 });
```

Comment:
Why do we want rows 300-400 if they're loaded already?

Reply:
When we try load next page (rows 400-500) width index less than page 350 (for `pageSize === 100`), we catch index bug. It loads nextPage instead curren when we fast scroll up.
We see at string 320, cache know about previous (200-300) and next page (400-500), and we load 400-500. Then, we get rows combination:
```
[200,
201,
...
300,
400,
401,
...
500,
401,
...
500]
```
For index less than `start - pageSize/2` we need to shift loading interval.

Reply:
Does it reproduce in non-infinite mode?